cobald_tests/daemon/core/test_config.py

```diff
@@ -13,6 +14,57 @@
 COBalDLoader.add_constructor(tag="!MockPool", constructor=yaml_constructor(MockPool))
 
 
+# Helpers for testing lazy/eager YAML evaluation
+# Since YAML defaults to lazy evaluation, the arguments available during evaluation
+# are not necessarily complete.
+class TagTracker:
+    """Helper to track the arguments supplied to YAML !Tags"""
+
+    def __init__(self, *args, **kwargs):
+        # the state of arguments *during* YAML evaluation
+        self.orig_args = copy.deepcopy(args)
+        self.orig_kwargs = copy.deepcopy(kwargs)
+        # the state of arguments *after* YAML evaluation
+        self.final_args = args
+        self.final_kwargs = kwargs
```

Comment:
This a bit weird on the first glance, but I have no really idea to make it clearer.